src/classes/services/CaseService.cls

```diff
@@ -0,0 +1,48 @@
+/**
+ * Created by Mariya on 25.01.2023.
+ */
+
+public inherited sharing class CaseService {
+
+    public static List<Case> getCasesWithNewOwnerId(List<Case> newCases, Map<Id, Case> oldCasesByIds) {
+
+        List<Case> casesWithNewOwnerId = new List<Case>();
+
+        for (Case updatedCase : newCases) {
+            if (updatedCase.OwnerId != oldCasesByIds.get(updatedCase.Id).OwnerId) {
+                casesWithNewOwnerId.add(updatedCase);
+            }
+        }
+
+        return casesWithNewOwnerId;
+    }
+
+    public static void createAppointmentResources(List<Case> newCases, Map<Id, Case> oldCasesByIds) {
+
+        List<Case> casesWithNewOwnerId = getCasesWithNewOwnerId(newCases, oldCasesByIds);
```

Comment:
MINOR
You could simplify this as well, e.g.
`
for (Case case : newCases) {

    if (isOwnerChanged(oldCasesByIds.get(case.Id), case) && isAppointmentRequestRT(case)) {

        filteredCasesByOwnerIds.put(case.OwnerId, case);

        filteredCasesIds.add(case.Id);

    }

}`